An AssertionFailure is reachable ("Should not get here without a service registry")

After I deploy contracts on a private network and launch `raiden` without `--service-registry-contract-address`, I [get an assertion failure](https://github.com/raiden-network/raiden/pull/5148#issuecomment-545918837).

`AssertionFailure` indicates a bug in the codebase, but in this case, `raiden` should request the user to specify `--service-registry-contract-address` or choose a different PFS URL.

The AssertionFailure mentioned is:

```
Traceback (most recent call last):
  File "/home/yoichi/priv_chain/env/bin/raiden", line 11, in <module>
    load_entry_point('raiden', 'console_scripts', 'raiden')()
  File "/home/yoichi/priv_chain/raiden/raiden/__main__.py", line 13, in main
    run(auto_envvar_prefix="RAIDEN")  # pylint: disable=no-value-for-parameter
  File "/home/yoichi/priv_chain/env/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/yoichi/priv_chain/env/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/yoichi/priv_chain/env/lib/python3.7/site-packages/click/core.py", line 1114, in invoke
    return Command.invoke(self, ctx)
  File "/home/yoichi/priv_chain/env/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/yoichi/priv_chain/env/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/yoichi/priv_chain/env/lib/python3.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/yoichi/priv_chain/raiden/raiden/ui/cli.py", line 576, in run
    app = runner.run()
  File "/home/yoichi/priv_chain/raiden/raiden/ui/runners.py", line 245, in run
    return self._start_services()
  File "/home/yoichi/priv_chain/raiden/raiden/ui/runners.py", line 92, in _start_services
    app_ = run_app(**self._options)
  File "/home/yoichi/priv_chain/raiden/raiden/ui/app.py", line 277, in run_app
    pathfinding_service_address=pathfinding_service_address,
  File "/home/yoichi/priv_chain/raiden/raiden/ui/startup.py", line 222, in setup_proxies_or_exit
    pathfinding_max_fee=config["services"]["pathfinding_max_fee"],
  File "/home/yoichi/priv_chain/raiden/raiden/network/pathfinding.py", line 251, in configure_pfs_or_exit
    assert service_registry, "Should not get here without a service registry"
AssertionError: Should not get here without a service registry
```

--- a/raiden/network/pathfinding.py
+++ b/raiden/network/pathfinding.py
@@ -248,7 +248,14 @@ def configure_pfs_or_exit(
     msg = "With PFS routing mode we shouldn't get to configure_pfs with pfs_address being None"
     assert pfs_url, msg
     if pfs_url == "auto":
-        assert service_registry, "Should not get here without a service registry"
+        if service_registry is None:
+            click.secho(
+                "Raiden was started with routing mode set to PFS, the pathfinding service address "
+                "set to 'auto' but no service registry address was given. Either specifically "
+                "provide a PFS address or provide a service registry address."
+            )
+            sys.exit(1)
+
         block_hash = service_registry.client.get_confirmed_blockhash()
         maybe_pfs_url = get_random_pfs(
             service_registry=service_registry,

